ui/test/nuxeo-collapsible.test.js

```diff
@@ -0,0 +1,158 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html, fixture, flush, isElementVisible, waitForAttrMutation, waitChanged, tap } from '@nuxeo/testing-helpers';
+import '../widgets/nuxeo-collapsible.js';
+import '../nuxeo-icons';
+/* eslint-disable no-unused-expressions */
+
+suite('nuxeo-collapsible', () => {
+  suite('Collapse behaviour', () => {
+    test('Should display nuxeo-collapsible when declaring the element', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible></nuxeo-collapsible>
+        `,
+      );
+      await flush();
+
+      expect(isElementVisible(nuxeoCollapsible.shadowRoot.querySelector('button'))).to.be.true;
+    });
+
+    test('Should show and hide content when heading is tapped', async () => {
+      const nuxeoCollapsible = await fixture(
+        html`
+          <nuxeo-collapsible>
+            <span slot="heading">Content</span>
+            <div>Collapsed content</div>
+          </nuxeo-collapsible>
+        `,
+      );
+      await flush();
+
+      const button = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const content = nuxeoCollapsible.shadowRoot.querySelectorAll('slot')[1].assignedNodes()[2];
```

Comment:
We have this kind of approach on several tests.
It works but it is not the easiest thing to read and even to code. I would suggest to create an helper to approach slots on tests.
Something that we can pass, for instance, which slot (by order, by some identifier) and also the inner content (by type or by some identifier).
WDYT? If you want, we can discuss this and create a ticket for tackling it. 😉

Reply:
I agree, it would be useful. I will create a separate ticket to handle that so we can move forward with this feature.